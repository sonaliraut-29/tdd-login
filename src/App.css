@import url("https://fonts.googleapis.com/css?family=Dosis:300");
/* variables */
:root {
  /* colors */
  --primary-color: #f44336;
  --primary-color-dark: color(#f44336 shade(12%));
  --primary-color-tint: color(#f44336 tint(70%));
  --secondary-color: #222;
  --secodnary-color-dark: color(#222 shade(12%));
  --secondary-color-tint: color(#222 tint(70%));

  /* width */
  --form-width: 16rem;
}

body {
  background-image: linear-gradient(140deg, #db6241 0%, #e55b8d 100%);
  height: 100vh;
  overflow: hidden;
  display: flex;
  justify-content: center;
  align-items: center;
}

.App {
  background: #27282d;
  height: 450px;
  width: 520px;
  margin: 0 auto;
  border-radius: 5px;
  box-shadow: 0px 4px 30px -5px rgba(0, 0, 0, 0.65);
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.input-container {
  position: relative;
  margin-bottom: 30px;
  width: 340px;
}

input,
button {
  font-family: "Dosis", sans-serif;
  letter-spacing: 1.3px;
}

input {
  box-sizing: border-box;
  background: transparent;
  width: 100%;
  border: none;
  border-bottom: 1px solid #626571;
  color: #ccced3;
  padding: 12px 6px 12px 36px;
  font-size: 19px;
  outline: none;
  caret-color: #e55b8d;
}

input:focus {
  border-bottom: 1px solid transparent;
  border-image: linear-gradient(140deg, #db6241 0%, #e55b8d 100%);
  border-image-slice: 1;
}

input:focus + i {
  color: #ccced3;
}

::placeholder {
  color: #ccced3;
  opacity: 0.4;
}

i {
  color: #626571;
  position: absolute;
  left: 6px;
  top: 50%;
  transform: translateY(-50%);
  transition: color 0.3s;
}

button {
  background: #33343b;
  border: none;
  border-radius: 30px;
  color: #ccced3;
  font-size: 16px;
  padding: 16px 36px;
  cursor: pointer;
  transition: all 0.3s;
  /* margin-top: 20px; */
  margin-bottom: -30px;
  box-shadow: 0px 4px 15px -5px rgba(0, 0, 0, 0.65);
  transition: all 0.3s;
}

button:hover {
  background: #3f4048;
  box-shadow: 0px 5px 20px -3px rgba(0, 0, 0, 0.65);
}

@media screen and (max-width: 620px) {
  .App {
    min-height: 300px;
    max-height: 420px;
    width: 90vw;
  }

  .input-container {
    width: 80%;
    margin-bottom: 40px;
  }

  button {
    margin-top: 10px;
    margin-bottom: -20px;
  }
}

.form_logo {
  font-family: "Lobster", cursive;
  font-size: 2.5rem;
  margin-bottom: 8px;
  color: white;
}

.form_logo span {
  color: #f44336;
  color: var(--primary-color);
}

.form_title {
  font-weight: bold;
  margin-bottom: 8px;
  color: white;
}

.form_title span {
  color: #f44336;
  color: var(--primary-color);
}

.error {
  color: red;
}
