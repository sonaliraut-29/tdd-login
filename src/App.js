import React, { useState } from "react";
import "./App.css";
import axios from "axios";

function App() {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  const [emailError, setEmailError] = useState(false);
  const [passwordError, setPasswordError] = useState(false);

  const [isLogin, setIsLogin] = useState(false);

  const handleInputUsername = (e) => {
    if (e.target.value) {
      setEmail(e.target.value);
      const emailErrorTemp =
        "" !== e.target.value &&
        !e.target.value.match(/^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/)
          ? true
          : false;
      setEmailError(emailErrorTemp);
    } else {
      setEmail("");
      setEmailError(true);
    }
  };

  const handleInputPassword = (e) => {
    if (e.target.value) {
      setPassword(e.target.value);
      const passwordErrorTemp =
        "" !== e.target.value && e.target.value.length <= 6 ? true : false;
      setPasswordError(passwordErrorTemp);
    } else {
      setPassword("");
      setPasswordError(true);
    }
  };
  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      if (email && password) {
        setIsLogin(true);
        await axios.post("/api/login", { email, password }).then((response) => {
          console.log(response);
        });
      }
    } catch (error) {
      console.log(error);
    }
  };
  return (
    <>
      {isLogin ? (
        <div>Login successful</div>
      ) : (
        <div className="App" data-testid="app">
          <div className="form_logo">
            Lo<span>g</span>o
          </div>
          <div className="form_title">
            Log<span>I</span>n
          </div>
          <div className="input-container" data-testid="input-username">
            <input
              type="text"
              placeholder="Username"
              onChange={handleInputUsername}
              value={email}
              data-testid="input-email"
            />
            <i className="zmdi zmdi-account zmdi-hc-lg"></i>
            {emailError && (
              <p data-testid="email-error" className="error">
                Email is required or invalid
              </p>
            )}
          </div>

          <div className="input-container">
            <input
              type="password"
              placeholder="Password"
              data-testid="input-password"
              onChange={handleInputPassword}
              value={password}
            />
            <i className="zmdi zmdi-lock zmdi-hc-lg"></i>
            {passwordError && (
              <p data-testid="password-error" className="error">
                Password is required or wrong
              </p>
            )}
          </div>

          <button
            type="button"
            data-testid="login-button"
            onClick={handleSubmit}
            disabled={emailError || passwordError ? true : false}
          >
            Log In
          </button>
        </div>
      )}
    </>
  );
}

export default App;
