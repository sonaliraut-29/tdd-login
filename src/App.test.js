import { render, screen, wait, fireEvent } from "@testing-library/react";
import App from "./App";

describe("<App />", () => {
  it("Should render the app login view", () => {
    render(<App />);
    expect(screen.getByTestId("app")).toMatchSnapshot();
  });

  it("Should render include input as username and password", () => {
    render(<App />);
    const loginButton = screen.getByTestId("login-button");

    expect(screen.getByTestId("input-username")).toBeInTheDocument();
    expect(screen.getByTestId("input-password")).toBeInTheDocument();
    expect(loginButton).toBeInTheDocument();
  });

  it("should show no error values when page is been loaded", () => {
    render(<App />);

    expect(screen.queryByTestId("password-error")).not.toBeInTheDocument();
    expect(screen.queryByTestId("email-error")).not.toBeInTheDocument();
  });

  it("should show error for invalid email", () => {
    render(<App />);

    fireEvent.change(screen.getByTestId("input-email"), {
      target: { value: "user@" },
    });

    expect(screen.getByTestId("email-error")).toHaveTextContent(
      "Email is required or invalid"
    );

    expect(screen.getByTestId("login-button")).toBeDisabled();
  });

  it("should show error for invalid password", () => {
    render(<App />);

    fireEvent.change(screen.getByTestId("input-password"), {
      target: { value: "user" },
    });

    expect(screen.getByTestId("password-error")).toHaveTextContent(
      "Password is required or wrong"
    );

    expect(screen.getByTestId("login-button")).toBeDisabled();
  });

  it("should show error for empty value of email", () => {
    render(<App />);

    fireEvent.change(screen.getByTestId("input-email"), {
      target: { value: " " },
    });

    expect(screen.getByTestId("email-error")).toHaveTextContent(
      "Email is required or invalid"
    );

    expect(screen.getByTestId("login-button")).toBeDisabled();
  });

  it("should show error for empty value of password", () => {
    render(<App />);

    fireEvent.change(screen.getByTestId("input-password"), {
      target: { value: " " },
    });

    expect(screen.getByTestId("password-error")).toHaveTextContent(
      "Password is required or wrong"
    );

    expect(screen.getByTestId("login-button")).toBeDisabled();
  });
});
