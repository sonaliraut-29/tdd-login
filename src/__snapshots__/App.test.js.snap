// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Should render the app login view 1`] = `
<div
  class="App"
  data-testid="app"
>
  <div
    class="form_logo"
  >
    Lo
    <span>
      g
    </span>
    o
  </div>
  <div
    class="form_title"
  >
    Log
    <span>
      I
    </span>
    n
  </div>
  <div
    class="input-container"
    data-testid="input-username"
  >
    <input
      data-testid="input-email"
      placeholder="Username"
      type="text"
      value=""
    />
    <i
      class="zmdi zmdi-account zmdi-hc-lg"
    />
  </div>
  <div
    class="input-container"
  >
    <input
      data-testid="input-password"
      placeholder="Password"
      type="password"
      value=""
    />
    <i
      class="zmdi zmdi-lock zmdi-hc-lg"
    />
  </div>
  <button
    data-testid="login-button"
    type="button"
  >
    Log In
  </button>
</div>
`;
